Utilities/Thread.cpp

```diff
@@ -2336,9 +2336,8 @@ bool thread_base::join(bool dtor) const
 	}
 
 	// Hacked for too sleepy threads (1ms) TODO: make sure it's unneeded and remove
-	const auto timeout = dtor && Emu.IsStopped() ? atomic_wait_timeout{1'000'000} : atomic_wait_timeout::inf;
+	const auto timeout = dtor && !thread_ctrl::get_current() ? atomic_wait_timeout{1'000'000} : atomic_wait_timeout::inf;
```

Comment:
same.

Reply:
Can you just keep this line the same? It does't stand in the way of the declared purpose of the PR. On the other hand, checking get_current() is a very weird.